dvc/utils/hydra.py

```diff
@@ -44,6 +48,10 @@ def compose_and_dump(
         dumper(output_file, OmegaConf.to_object(cfg))
     else:
         Path(output_file).write_text(OmegaConf.to_yaml(cfg), encoding="utf-8")
+    contents = Path(output_file).read_text(encoding="utf-8")
```

Comment:
Minor suggestion: lets store contents in a variable, and it to write texto and the log, so we don't have to read the file?

Reply:
It would be better, but I can't see it being that easy for non-yaml configs since the dumping utils are so tightly coupled with the filepath. Am I missing an easy way to do it?

Reply:
@dberenbaum, you can print the `cfg` object itself, no? It's a dict, but for debugging, it's better than having multiline output.

I'll suggest making it a `trace` instead of a debug for reducing noise, and seems like it's a trace than a debug.

Reply:
I thought it was more obvious and useful to show it the same way it's written to the file, but I see how you have it set for interpolation here:

https://github.com/iterative/dvc/blob/7f6890ff25bb43e3e0acc87f072f5d8d5dc28534/dvc/parsing/__init__.py#L280-L282

I think it makes sense to try to make it consistent, so I'll make mine the same (trace-level output showing the raw python object). I think we should be documenting more of these in the user guide. I didn't realize until now that there was a way to see the resolved stages.

Reply:
Both of those cases I consider it belonging to debug and not trace, but I guess this is one of those endless debates in software where people have different contexts and experiences 😅 It is something would help users debug their issues.

Not to reinforce my opinion, but in the 2-min Google search I did, these cases match the definitions of what people say it should be in DEBUG

Reply:
I think documenting these and showing how to use them is more important than the output format or log level.

Reply:
> I think documenting these and showing how to use them is more important than the output format or log level.

Indeed. What I mean is that we will be requiring people to enable TRACE which will be quite verbose + most likely not containing info they can use for their problem.
So that was just my reasoning. These clearly feels like useful for user-facing, so makes sense to put it in debug